src/components/Authorize/Authorize.js

```diff
@@ -11,26 +11,54 @@ const Authorize = () => {
       .substring(1)
       .split('access_token=')[1]
       .split('&')[0];
-    var options = {
-      url: 'https://api.spotify.com/v1/me',
-      headers: { Authorization: 'Bearer ' + access_token },
-      json: true,
-    };
+    console.log(access_token);
+    let state = window.location.hash
+      .substring(1)
+      .split('state=')[1];
+    if(state === localStorage.getItem("stateValue")){
+     var options = {
+        url: 'https://api.spotify.com/v1/me',
+        headers: { Authorization: 'Bearer ' + access_token },
+        json: true,
+      };
 
-    // use the access token to access the Spotify Web API
-    request.get(options, function (error, response, body) {
-      console.log(body);
+      // use the access token to access the Spotify Web API
+      request.get(options, function (error, response, body) {
+        if(error) {
+          localStorage.removeItem("stateValue")
+          return setUserData({
+            ...userData,
+            isLoggedIn: "error",
+            data: {}
+          });
+          
+        }
+        localStorage.setItem("token", access_token);
+        setUserData({
+          ...userData,
+          isLoggedIn: true,
+          data: body,
+        });
+      });
+    } else {
       setUserData({
-        isLoggedIn: true,
-        data: body,
+        ...userData,
+        isLoggedIn: "error",
+        data: {}
       });
-    });
+      localStorage.removeItem("stateValue");
+    }
   }, []);
 
   return (
     <div>
-      {userData.isLoggedIn && <Navigate to='/home' />}
-      <h1>Hello</h1>
+      {userData.isLoggedIn === true ? 
+        <Navigate to='/home' /> :
+        userData.isLoggedIn === "error" ? 
+        <Navigate to="/login" /> : 
```

Comment:
This redirect doesn't work on page refreshes due to reasons:

1. Default value of `userData.isLoggedIn` set to `false` in `App.js` at line 21. 
2.  `Authorize` component only used in first time login through login page, if I refresh the page in a different route like `/home` it doesn't work at all.